### PR TITLE
stop replacing bivariant args with `'static` when computing closure requirements

### PR DESCRIPTION
**Pull request by lcnr**
It is unnecessary, these get constrained when checking that the opaque type is well-formed.

It also results in the opaque type no longer being well formed. If you've got `fn foo<'a>() -> impl Sized + 'a` the opaque is `type Opaque<'a, 'aDummy> where 'a: 'aDummy, 'aDummy: 'a` where `'aDummy`  is bivariant. If we call `foo::<'b>()`  inside of a closure and its return type ends up in a type test, we start out with the WF `Opaque<'b, 'b>`, and then replace the bivariant `'b` with `'static`. `Opaque<'b, 'static>` is no longer well-formed. Given how these type tests are used, I don't think this caused any practical issues.

r? types

### CODE REVIEW

File: compiler/rustc_borrowck/src/region_infer/mod.rs [hunk 1]
```diff
@@ -1008,50 +1008,31 @@ impl<'tcx> RegionInferenceContext<'tcx> {
...
-            // This is slightly too conservative. To show T: '1, given `'2: '1`
-            // and `'3: '1` we only need to prove that T: '2 *or* T: '3, but to
-            // avoid potential non-determinism we approximate this by requiring
-            // T: '1 and T: '2.
-            for upper_bound in non_local_ub {
-                debug_assert!(self.universal_regions().is_universal_region(upper_bound));
-                debug_assert!(!self.universal_regions().is_local_free_region(upper_bound));
-
+            if !self.universal_regions().is_local_free_region(ur) {
+                debug_assert!(!self.universal_regions().is_local_free_region(ur));
```
Comment (lqd) [inline compiler/rustc_borrowck/src/region_infer/mod.rs:1021]: meow

File: compiler/rustc_borrowck/src/region_infer/mod.rs [hunk 2]
```diff
@@ -1013,94 +1008,45 @@ impl<'tcx> RegionInferenceContext<'tcx> {
...
+                    outlived_free_region: ur,
+                    blame_span,
                     category: ConstraintCategory::Boring,
                 };
-                debug!("try_promote_type_test: pushing {:#?}", requirement);
+                debug!(?requirement, ?ur, "adding closure requirement");
                 propagated_outlives_requirements.push(requirement);
+                found_candidate = true;
+            } else {
+                debug!(?ur, "skippiung local universal");
```
Comment (lqd) [inline compiler/rustc_borrowck/src/region_infer/mod.rs:1032]:
```suggestion
                debug!(?ur, "skipping local universal");
```

Comment (lcnr) [reply compiler/rustc_borrowck/src/region_infer/mod.rs]: unfortunately reverting that commit :<

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**